### PR TITLE
DE-325 Update API v1 and v2 03-28-2019

### PR DESCRIPTION
**Pull request by adducci**
Updates include:
- SearchFilter added to Search Properties
- Response removed from SystemUsersResetMfa endpoint
- List/ Create Provider Admin endpoints added

### CODE REVIEW

File: v1/applications_api.go [hunk 1]
```diff
@@ -27,7 +27,7 @@ type ApplicationsApiService service
 
 
 /* ApplicationsApiService Applications
- The endpoint returns all your SSO / SAML Applications.  #### Sample Request &#x60;&#x60;&#x60; curl -X GET https://console.jumpcloud.com/api/applications \\   -H &#39;Accept: application/json&#39; \\   -H &#39;Content-Type: application/json&#39; \\   -H &#39;x-api-key: {API_KEY}&#39;  &#x60;&#x60;&#x60;
+ The endpoint returns all your SSO / SAML Applications.  #### Sample Request &#x60;&#x60;&#x60; curl -X GET https://console.jumpcloud.com/api/applications \\   -H &#39;Accept: application/json&#39; \\   -H &#39;Content-Type: application/json&#39; \\   -H &#39;x-api-key: {API_KEY}&#39;   &#x60;&#x60;&#x60;
```
Comment (lsi1verthorn) [inline v1/applications_api.go]: extra space?

File: v1/command_results_api.go [hunk 2]
```diff
@@ -28,7 +28,7 @@ type CommandResultsApiService service
 
 
 /* CommandResultsApiService Delete a Command result
- This endpoint deletes a specific command result.  #### Sample Request &#x60;&#x60;&#x60; curl -X GET https://console.jumpcloud.com/api/commandresults/{CommandID} \\   -H &#39;Accept: application/json&#39; \\   -H &#39;Content-Type: application/json&#39; \\   -H &#39;x-api-key: {API_KEY}&#39;   &#x60;&#x60;&#x60;&#x60;
+ This endpoint deletes a specific command result.  #### Sample Request &#x60;&#x60;&#x60; curl -X GET https://console.jumpcloud.com/api/commandresults/{CommandID} \\   -H &#39;Accept: application/json&#39; \\   -H &#39;Content-Type: application/json&#39; \\   -H &#39;x-api-key: {API_KEY}&#39;    &#x60;&#x60;&#x60;&#x60;
```
Comment (lsi1verthorn) [inline v1/command_results_api.go]: seems like there's an extra space that was added?

File: v2/api/swagger.yaml [hunk 3]
```diff
@@ -4246,6 +4247,176 @@ paths:
...
+      parameters:
+      - name: "provider_id"
+        in: "path"
+        required: true
+        type: "string"
+        x-exportParamName: "ProviderId"
+      - name: "fields"
+        in: "query"
+        description: "The comma separated fields included in the returned records.\n\
+          If omitted the default list of fields will be returned.\n"
```
Comment (lsi1verthorn) [inline v2/api/swagger.yaml]: `If omitted,`

File: v2/api/swagger.yaml [hunk 4]
```diff
@@ -4246,6 +4247,176 @@ paths:
...
+        mock:
+          enabled: false
+          dynamic: false
+          statusCode: 200
+    post:
+      tags:
+      - "Providers"
+      summary: "Create a new Provider Administrator"
+      description: "This endpoint allows you to create a provider administrator. You\
+        \ must be associated to given provider to use this route.\n\n**Sample Request**\n\
```
Comment (lsi1verthorn) [inline v2/api/swagger.yaml]: `to the provider`? `to the given provider`?

File: v2/docs/ProvidersApi.md [hunk 5]
```diff
@@ -0,0 +1,96 @@
+# \ProvidersApi
```
Comment (lsi1verthorn) [inline v2/docs/ProvidersApi.md]: should the `\` be there?

File: v2/docs/ProvidersApi.md [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+------------- | ------------- | -------------
+[**ProvidersListAdministrators**](ProvidersApi.md#ProvidersListAdministrators) | **Get** /providers/{provider_id}/administrators | List Provider Administrators
+[**ProvidersPostAdmins**](ProvidersApi.md#ProvidersPostAdmins) | **Post** /providers/{provider_id}/administrators | Create a new Provider Administrator
+
+
+# **ProvidersListAdministrators**
+> InlineResponse200 ProvidersListAdministrators(ctx, providerId, contentType, accept, optional)
+List Provider Administrators
+
+This endpoint returns a list of the Administrators associated with the Provider.  #### Sample Request ``` curl -X GET https://console.jumpcloud.com/api/v2/providers/{ProviderID}/administrators \\   -H 'Accept: application/json' \\   -H 'Content-Type: application/json' \\   -H 'x-api-key: {API_KEY}' ```
```
Comment (lsi1verthorn) [inline v2/docs/ProvidersApi.md]: Should the `#### Sample...` start on a new line?

File: v2/docs/ProvidersApi.md (hunk 5 above)
Comment (adducci) [reply v2/docs/ProvidersApi.md]: Yes it marks it as a heading (ProvidersApi section vs the ProvidersListAdmin member of the section)

File: v1/applications_api.go (hunk 1 above)
Comment (adducci) [reply v1/applications_api.go]: The extra spaces are just part of the automatic generation

File: v2/docs/ProvidersApi.md (hunk 6 above)
Comment (adducci) [reply v2/docs/ProvidersApi.md]: That's also just how it's automatically generated.

Comment (lsi1verthorn) [reply v2/docs/ProvidersApi.md]: The reason I asked was it seemed like it was on a newline in other places but just must be how the spacing worked out.

**Pull request merged**